lib/alexa_request_verifier/verifier.rb

```diff
@@ -4,6 +4,121 @@
 module AlexaRequestVerifier
   # A namespace for all of our verifiers to live under
   # @since 0.1
-  module Verifier
+  class Verifier
+    attr_accessor :configuration
+
+    # Create a new AlexaRequestVerifier::Verifier object
+    #
+    # @yield the configuration block
+    # @yieldparam config [AlexaRequestVerifier::Configuration] the configuration object
+    def initialize
+      @configuration = AlexaRequestVerifier::Configuration.new
+
+      yield @configuration if block_given?
+    end
+
+    # Validate a request object from Sinatra.
+    # Raise an error if it is not valid.
+    #
+    # @param [Sinatra::Request] request a Sinatra HTTP Request
+    #
+    # @raise [AlexaRequestVerifier::InvalidCertificateURIError]
+    #   there was a problem validating the certificate URI from your request
+    #
+    # @return [nil] will always return nil
+    def valid!(request)
+      signature_certificate_url = request.env['HTTP_SIGNATURECERTCHAINURL']
+
+      AlexaRequestVerifier::Verifier::CertificateURIVerifier.valid!(signature_certificate_url) if @configuration.verify_uri?
+
+      raw_body = request.body.read
+      request.body.rewind
+
+      check_that_request_is_timely(raw_body) if @configuration.verify_timeliness?
+
+      check_that_request_is_valid(signature_certificate_url, request, raw_body)
+
+      true
+    end
+
+    # Validate a request object from Sinatra.
+    # Return a boolean.
+    #
+    # @param [Sinatra::Request] request a Sinatra HTTP Request
+    # @return [Boolean] is the request valid?
+    def valid?(request)
+      begin
+        valid!(request)
+      rescue AlexaRequestVerifier::BaseError => e
+        puts e
```

Comment:
Should this be here?

Reply:
Well, that's up for discussion to be fair - I wasn't sure if showing the reason it was returning false would be of use?

Otherwise it's just a 'black box' and I think it would be harder to debug?